### PR TITLE
metrics: add statement deadlock detect duration

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve? 
If it is sth wrong with the deadlock detector, e.g. very slow, we need to know it.

### What is changed and how it works?
Add statement deadlock detect duration.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: metrics/session.go [hunk 1]
```diff
@@ -90,6 +90,16 @@ var (
 			Help:      "Bucketed histogram of a transaction execution duration, including retry.",
 			Buckets:   prometheus.ExponentialBuckets(0.001, 2, 20), // 1ms ~ 1049s
 		}, []string{LblSQLType, LblType})
+
+	StatementDeadlockDetectDuration = prometheus.NewHistogram(
+		prometheus.HistogramOpts{
```
Comment (lonng) [inline metrics/session.go]: Do we need to add a label to distinct the `point_get` duration?

**Comment by cfzjywxk** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**